### ISSUE TITLE
tcl tk failure with R extension tkrplot

### ISSUE THREAD

**Issue by verdurin** [hpcugent/easybuild-easyconfigs#2749]
This is with `R-3.2.3-foss-2015b.eb` from `2.7.0`:

```
* installing *source* package ‘tkrplot’ ...
** package ‘tkrplot’ successfully unpacked and MD5 sums checked
checking for gcc... gcc
checking for C compiler default output... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables...
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for X... libraries , headers
checking for gethostbyname... yes
checking for connect... yes
checking for remove... yes
checking for shmat... yes
checking for IceConnectionNumber in -lICE... yes
checking for tclConfig.sh... no
checking for tclConfig.sh in library (sub)directories... no
checking for tkConfig.sh... no
checking for tkConfig.sh in library (sub)directories... no
checking for tcl.h... yes
checking for tk.h... yes
configure: creating ./config.status
config.status: creating src/Makevars
** libs
gcc -std=gnu99 -I/srv/data/sw/eb/software/R/3.2.3-foss-2015b/lib64/R/include -DNDEBUG  -I/srv/data/sw/eb/software/OpenBLAS/0.2.14-GNU-4.9.3-2.25-LAPACK-3.5.0/include -I/srv/data/sw/eb/software/ScaLAPACK/2.0.2-gompi-2015b-OpenBLAS-0.2.14-LAPACK-3.5.0/include -I/srv/data/sw/eb/software/FFTW/3.3.4-gompi-2015b/include -I/srv/data/sw/eb/software/libreadline/6.3-foss-2015b/include -I/srv/data/sw/eb/software/ncurses/5.9-foss-2015b/include -I/srv/data/sw/eb/software/libpng/1.6.18-foss-2015b/include -I/srv/data/sw/eb/software/libjpeg-turbo/1.4.1-foss-2015b/include -I/srv/data/sw/eb/software/Java/1.8.0_72/include -I/srv/data/sw/eb/software/Tcl/8.6.4-foss-2015b/include -I/srv/data/sw/eb/software/Tk/8.6.4-foss-2015b-no-X11/include -I/srv/data/sw/eb/software/cURL/7.45.0-foss-2015b/include -I/srv/data/sw/eb/software/libxml2/2.9.2-foss-2015b/include -I/srv/data/sw/eb/software/GDAL/2.0.1-foss-2015b/include -I/srv/data/sw/eb/software/PROJ/4.8.0-foss-2015b/include -I/srv/data/sw/eb/software/cairo/1.14.2-foss-2015b/include    -fpic  -O2 -march=native  -c tcltkimg.c -o tcltkimg.o
gcc -std=gnu99 -shared -L/srv/data/sw/eb/software/R/3.2.3-foss-2015b/lib64/R/lib -L/srv/data/sw/eb/software/GCC/4.9.3-binutils-2.25/lib64 -L/srv/data/sw/eb/software/GCC/4.9.3-binutils-2.25/lib -L/srv/data/sw/eb/software/OpenBLAS/0.2.14-GNU-4.9.3-2.25-LAPACK-3.5.0/lib -L/srv/data/sw/eb/software/ScaLAPACK/2.0.2-gompi-2015b-OpenBLAS-0.2.14-LAPACK-3.5.0/lib -L/srv/data/sw/eb/software/FFTW/3.3.4-gompi-2015b/lib -L/srv/data/sw/eb/software/libreadline/6.3-foss-2015b/lib -L/srv/data/sw/eb/software/ncurses/5.9-foss-2015b/lib -L/srv/data/sw/eb/software/libpng/1.6.18-foss-2015b/lib -L/srv/data/sw/eb/software/libjpeg-turbo/1.4.1-foss-2015b/lib -L/srv/data/sw/eb/software/Java/1.8.0_72/lib -L/srv/data/sw/eb/software/Tcl/8.6.4-foss-2015b/lib -L/srv/data/sw/eb/software/Tk/8.6.4-foss-2015b-no-X11/lib -L/srv/data/sw/eb/software/cURL/7.45.0-foss-2015b/lib -L/srv/data/sw/eb/software/libxml2/2.9.2-foss-2015b/lib -L/srv/data/sw/eb/software/GDAL/2.0.1-foss-2015b/lib -L/srv/data/sw/eb/software/PROJ/4.8.0-foss-2015b/lib -L/srv/data/sw/eb/software/cairo/1.14.2-foss-2015b/lib -o tkrplot.so tcltkimg.o -L/srv/data/sw/eb/software/R/3.2.3-foss-2015b/lib64/R/lib -lR
installing to /srv/data/sw/eb/software/R/3.2.3-foss-2015b/lib64/R/library/tkrplot/libs
** R
** preparing package for lazy loading
Warning: S3 methods ‘as.character.tclObj’, ‘as.character.tclVar’, ‘as.double.tclObj’, ‘as.integer.tclObj’, ‘as.logical.tclObj’, ‘as.raw.tclObj’, ‘print.tclObj’, ‘[[.tclArray’, ‘[[<-.tclArray’, ‘$.tclArray’, ‘$<-.tclArray’, ‘names.tclArray’, ‘names<-.tclArray’, ‘length.tclArray’, ‘length<-.tclArray’, ‘tclObj.tclVar’, ‘tclObj<-.tclVar’, ‘tclvalue.default’, ‘tclvalue.tclObj’, ‘tclvalue.tclVar’, ‘tclvalue<-.default’, ‘tclvalue<-.tclVar’, ‘close.tkProgressBar’ were declared in NAMESPACE but not found
Error : .onLoad failed in loadNamespace() for 'tcltk', details:
  call: fun(libname, pkgname)
  error: Tcl/Tk support is not available on this system
Error : package ‘tcltk’ could not be loaded
ERROR: lazy loading failed for package ‘tkrplot’
 (at easybuild/tools/run.py:397 in parse_cmd_output)
```

**Comment by sjtbham** [comment]
I noticed this on one of our builds, looking at previous issues, I added the tk-devel and tcl-devel packages and this then built. However, as there is a dev on Tcl and Tk, I'd expect those to provide the correct files?

**Comment by verdurin** [comment]
Yes, there is a deeper problem here with the `Tcl` and `Tk` easyconfigs.

**Comment by stachon** [comment]
+1 EB 2.8.0

**Comment by stachon** [comment]
you need to add " --with-tcltk --with-tcl-config=$EBROOTTCL/lib/tclConfig.sh --with-tk-config=$EBROOTTK/lib/tkConfig.sh" to configopts for R. I will submit a PR

**Comment by verdurin** [comment]
@stachon - that did work, so I'll generate a PR with the change.

**Comment by stachon** [comment]
this is probably related to #2261 #3079

**Comment by backelj** [comment]
I got the same issue with R-3.3.1-intel-2016b.eb :-(

It's not the first issue with the easyconfigs for toolchain 2016b, and its sad to see that these pass the jenkins testing.

Off course, this is probably due to the fact that system packages are installed and that EasyBuild is not capable of ignoring (all of) the installed libs.

**Comment by boegel** [comment]
@verdurin ping on the PR to add those configure flags?

**Comment by boegel** [comment]
I opened a PR for the R easyblock to add the missing `--with-tcl-config` and `--with-tk-config` configure options, see https://github.com/hpcugent/easybuild-easyblocks/pull/995 .

**Comment by boegel** [comment]
considering this fixed with the updated easyblock https://github.com/hpcugent/easybuild-easyblocks/pull/995 that will be included in EB v2.9.0...
